### PR TITLE
Fixed the missing min_sites and required_sites for Job.

### PR DESCRIPTION
**Pull request by yhwen**

### CODE REVIEW

**Comment by yhwen** [comment]
> 

Interesting Pycharm and python3 -m isort/black results are different now.

File: nvflare/apis/job_def.py [hunk 1]
```diff
@@ -144,16 +144,18 @@ def job_from_meta(meta: dict) -> Job:
     """Converts information in meta into a Job object.
 
     Args:
-        meta: dict of meta information
+        meta: dict of meta information(
```
Comment (YuanTingHsieh) [inline nvflare/apis/job_def.py:147]: typo extra "("

File: nvflare/apis/job_def.py [hunk 2]
```diff
@@ -144,16 +144,18 @@ def job_from_meta(meta: dict) -> Job:
...
-        meta.get(JobMetaKey.STUDY_NAME),
-        meta.get(JobMetaKey.RESOURCE_SPEC),
-        meta.get(JobMetaKey.DEPLOY_MAP),
-        meta,
+        job_id=meta.get(JobMetaKey.JOB_ID),
+        study_name=meta.get(JobMetaKey.STUDY_NAME),
+        resource_spec=meta.get(JobMetaKey.RESOURCE_SPEC),
+        deploy_map=meta.get(JobMetaKey.DEPLOY_MAP),
+        meta=meta,
+        min_sites=meta.get(JobMetaKey.MIN_CLIENTS),
```
Comment (YuanTingHsieh) [inline nvflare/apis/job_def.py:158]: If JobMetaKey.MIN_CLIENTS in missing, then this going to set min_sites = None.

We should change this line to `meta.get(JobMetaKey.MIN_CLIENTS, 1)`

File: nvflare/private/fed/client/client_engine.py [hunk 3]
```diff
@@ -74,6 +79,9 @@ def do_validate(self, req: Message):
 
         return "validate process started."
 
+    def new_context(self) -> FLContext:
+        return self.fl_ctx_mgr.new_context()
+
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/client/client_engine.py:84]: Why do we need this function here?

File: nvflare/apis/job_def.py (hunk 1 above)
Comment (yhwen) [reply nvflare/apis/job_def.py]: fixed.

File: nvflare/apis/job_def.py (hunk 2 above)
Comment (yhwen) [reply nvflare/apis/job_def.py]: fixed.

File: nvflare/private/fed/client/client_engine.py (hunk 3 above)
Comment (yhwen) [reply nvflare/private/fed/client/client_engine.py:84]: This is fully related to this bug. But we need this change for the client_engine.new_context().

**Review by YuanTingHsieh — APPROVED** [review]
LGTM

**Pull request merged**